Rely on default pylint configuration

ferrarimarco opened the pull request:
# Proposed changes

Remove configuration values from pylint configuration file in case they match with their defaults.

Fix #5220.

## Readiness checklist

In order to have this pull request merged, complete the following tasks.

### Pull request author tasks

- [x] I included all the needed documentation for this change.
- [x] I provided the necessary tests.
- [x] I squashed all the commits into a single commit.
- [x] I followed the [Conventional Commit v1.0.0 spec](https://www.conventionalcommits.org/en/v1.0.0/).
- [x] I wrote the necessary upgrade instructions in the [upgrade guide](../docs/upgrade-guide.md).
- [x] If this pull request is about and existing issue,
  I added the `Fix #ISSUE_NUMBER` label to the description of the pull request.

### Super-linter maintainer tasks

- [x] Label as `breaking` if this change breaks compatibility with the previous released version.
- [x] Label as either: `automation`, `bug`, `documentation`, `enhancement`, `infrastructure`.

Hanse00 commented:
On one hand, I see the point of deduplication. On the other, I can imagine the argument for creating this in the first place might have been: Relying on defaults also means they can change underneath you without noticing.

I don't have a strong leaning towards either. @zkoppert any input you want to provide?

ferrarimarco commented:
Thanks for the review!

In the case of Pylint, anything that users don't set, falls back on default values. The configuration file I removed configured values that matched with default ones, leading to duplication. That's why I removed most values.

If we keep this file, we would need to maintain it as the pylint configuration evolves. While working on this, I realized that some values that we set included obsolete values, while some others were moved to different sections. I think there's relatively little value in keeping a configuration file that duplicates the default values when the tool already provides a fallback mechanism to default value.

Hanse00 reviewed (approved):
Works for me, if Zack feels otherwise we can always revert.
